tests/system/test_cluster/test_agent_files_deletion/test_agent_files_deletion.py

```diff
@@ -1,46 +1,76 @@
 # Copyright (C) 2015-2021, Wazuh Inc.
 # Created by Wazuh, Inc. <info@wazuh.com>.
 # This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
-
+import os
+import re
 from os.path import join, dirname, abspath
-from time import sleep
+from time import time, sleep
 
 import pytest
-
-from wazuh_testing.tools import WAZUH_PATH
+from wazuh_testing.tools import WAZUH_PATH, WAZUH_LOGS_PATH
+from wazuh_testing.tools.monitoring import HostMonitor
 from wazuh_testing.tools.system import HostManager
 
 master_host = 'wazuh-master'
 worker_host = 'wazuh-worker2'
 agent_host = 'wazuh-agent3'
+local_path = os.path.dirname(os.path.abspath(__file__))
+messages_path = os.path.join(local_path, 'data/messages.yml')
+script_path = os.path.join(re.sub(r'^.*?wazuh-qa', '/wazuh-qa', local_path), 'data/get_wdb_agent.py')
+
+tmp_path = os.path.join(local_path, 'tmp')
 managers_hosts = [master_host, worker_host]
 inventory_path = join(dirname(dirname(dirname(abspath(__file__)))), 'provisioning', 'basic_cluster', 'inventory.yml')
 host_manager = HostManager(inventory_path)
-time_to_sync = 60
+while_time = 5
+time_to_sync = 20
+time_to_agent_reconnect = 180
 
 # Each file should exist in all hosts specified in 'hosts'.
 files = [{'path': join(WAZUH_PATH, 'queue', 'rids', '{id}'), 'hosts': managers_hosts},
          {'path': join(WAZUH_PATH, 'queue', 'agent-groups', '{id}'), 'hosts': managers_hosts},
          {'path': join(WAZUH_PATH, 'queue', 'diff', '{name}'), 'hosts': [worker_host]},
          {'path': join(WAZUH_PATH, 'queue', 'db', '{id}.db'), 'hosts': [worker_host]}]
-db_queries = ["select * from agent where id={id}",
-              "select * from belongs where id_agent={id}"]
+
+queries = ['global sql select * from agent where id={id}',
+           'global sql select * from belongs where id_agent={id}']
 
 
-@pytest.fixture(scope='function')
-def register_agent():
-    """Restart the removed agent to trigger auto-enrollment."""
-    yield
-    host_manager.get_host(agent_host).ansible('command', f'service wazuh-agent restart', check=False)
+def agent_healthcheck(master_token):
+    """Check if the agent is active and reporting."""
+
+    timeout = time() + time_to_agent_reconnect
+    while True:
+        response = host_manager.make_api_call(host=master_host, method='GET', token=master_token,
+                                              endpoint='/agents?status=active')
+        assert response['status'] == 200, 'Failed when trying to get the active agents'
+        if int(response['json']['data']['total_affected_items']) == 4:
+            break
+        elif time() > timeout:
+            print("The agent 'wazuh-agent3' is not 'Active' yet.")
```

Comment:
We are not exiting the while, so when we can't find the `wazuh-agent3` active, the test will keep printing `The agent 'wazuh-agent3' is not 'Active' yet.` forever.